driver/src/price_estimation/mod.rs

```diff
@@ -32,19 +37,29 @@ pub struct PriceOracle {
     /// whitelisted tokens get their prices estimated.
     tokens: TokenData,
     /// The price source to use.
-    ///
-    /// Note that currently only one price source is supported, but more price
-    /// source are expected to be added in the future.
     source: Box<dyn PriceSource>,
 }
 
 impl PriceOracle {
     /// Creates a new price oracle from a token whitelist data.
-    pub fn new(tokens: TokenData) -> Result<Self> {
-        Ok(PriceOracle::with_source(
-            tokens,
-            AveragePriceSource::new(KrakenClient::new()?, DexagClient::new()?),
-        ))
+    pub fn new(tokens: TokenData, update_interval: Duration) -> Result<Self> {
+        Ok(if tokens.is_empty() {
+            // If there are no tokens we don't need to create a real price
+            // source. This avoids sending requests in the Dexag constructor.
+            struct EmptyPriceSource {};
+            impl PriceSource for EmptyPriceSource {
```

Comment:
While it's cool that we can inline implementations like this, I'd prefer to lift it outside and make it a top level implementation to keep this function more readable.

Reply:
On the other hand if it is only needed in one function then it might be more readable to have it there. I don't mind either way so I'll move it out.

Reply:
Actually this won't be needed with the lazy initialization of dexag PR.